src/binaryen-c.h

```diff
@@ -707,6 +707,19 @@ typedef struct BinaryenBufferSizes {
 // @returns how many bytes were written. This will be less than or equal to outputSize
 BinaryenBufferSizes BinaryenModuleWriteWithSourceMap(BinaryenModuleRef module, const char* url, char* output, size_t outputSize, char* sourceMap, size_t sourceMapSize);
 
+// Result structure of BinaryenModuleToBinary. Allocated buffers for both the output and
+// the source map must be freed manually.
+typedef struct BinaryenModuleToBinaryResult {
+  void* binary;
+  size_t binaryBytes;
+  char* sourceMap;
+} BinaryenModuleToBinaryResult;
+
+// Serializes a module into binary form, optionally including its source map if
+// sourceMapUrl has been specified. Uses the currently set global debugInfo option.
+// Allocated buffers for both the output and the source map must be freed manually.
+BinaryenModuleToBinaryResult BinaryenModuleToBinary(BinaryenModuleRef module, const char* sourceMapUrl);
```

Comment:
I'm not too sure about the name `BinaryenModuleToBinary`. My inclination is it should be more similar to the existing `BinaryenModuleWrite` maybe with a suffix that indicates how it differs. `BinaryenModuleWriteExtended`? `BinaryenModuleWrite2`?

Alternatively, maybe `BinaryenModuleAllocateAndWrite`?

Reply:
BinaryenModuleAllocateAndWrite seems good